src/lib/pipeline/ReorderBuffer.cc

```diff
@@ -60,6 +63,7 @@ void ReorderBuffer::commitMicroOps(uint64_t insnId) {
       }
       if (!validForCommit) return;
 
+      assert(firstOp != UINT64_MAX && "firstOp hasn't been populated");
```

Comment:
In the current form `firstOp` won't be populated without `foundFirstInstance` being set to true. If it were false then you would never get to the assertion. We would hope that none of our assertions get triggered if our code is correct, so that is expected, but this probably is a bit overkill can remove if you want

Reply:
It provides confidence and something to catch if lofic changes in the future